src/commands/auth/device/login.ts

```diff
@@ -5,85 +5,98 @@
     }),
-    instanceurl: flags.url({
+    'instance-url': Flags.url({
       char: 'r',
-      description: commonMessages.getMessage('instanceUrl'),
+      summary: commonMessages.getMessage('instanceUrl'),
+      deprecateAliases: true,
+      aliases: ['instanceurl'],
     }),
-    setdefaultdevhubusername: flags.boolean({
+    'set-default-dev-hub': Flags.boolean({
       char: 'd',
-      description: commonMessages.getMessage('setDefaultDevHub'),
+      summary: commonMessages.getMessage('setDefaultDevHub'),
+      deprecateAliases: true,
+      aliases: ['setdefaultdevhub'],
     }),
-    setdefaultusername: flags.boolean({
+    'set-default': Flags.boolean({
       char: 's',
-      description: commonMessages.getMessage('setDefaultUsername'),
+      summary: commonMessages.getMessage('setDefaultUsername'),
+      deprecateAliases: true,
+      aliases: ['setdefaultusername'],
     }),
-    setalias: flags.string({
+    alias: Flags.string({
       char: 'a',
-      description: commonMessages.getMessage('setAlias'),
+      summary: commonMessages.getMessage('setAlias'),
+      deprecateAliases: true,
+      aliases: ['setalias'],
     }),
-    disablemasking: flags.boolean({
-      description: commonMessages.getMessage('disableMasking'),
+    'disable-masking': Flags.boolean({
+      summary: commonMessages.getMessage('disableMasking'),
       hidden: true,
+      deprecateAliases: true,
+      aliases: ['disablemasking'],
     }),
   };
 
   public async run(): Promise<AuthFields> {
-    if (await Prompts.shouldExitCommand(this.ux, false)) return {};
+    const { flags } = await this.parse(Login);
+    if (await this.shouldExitCommand(false)) return {};
 
     const oauthConfig: OAuth2Config = {
-      loginUrl: await Common.resolveLoginUrl(get(this.flags.instanceurl, 'href', null) as Optional<string>),
-      clientId: this.flags.clientid as string,
+      loginUrl: await Common.resolveLoginUrl(get(flags.instanceurl, 'href', null) as Optional<string>),
+      clientId: flags.clientid as string,
     };
 
-    if (this.flags.clientid) {
-      oauthConfig.clientSecret = await Prompts.askForClientSecret(this.ux, this.flags.disablemasking as boolean);
+    if (flags.clientid) {
+      oauthConfig.clientSecret = await this.askForClientSecret(flags['disable-masking']);
     }
 
     const deviceOauthService = await DeviceOauthService.create(oauthConfig);
     const loginData = await deviceOauthService.requestDeviceLogin();
 
-    if (this.flags.json) {
-      this.ux.logJson(loginData);
+    if (flags.json) {
+      this.styledJSON(loginData);
     } else {
-      this.ux.styledHeader(messages.getMessage('actionRequired'));
-      this.ux.log(messages.getMessage('enterCode'), loginData.user_code, loginData.verification_uri);
-      this.ux.log();
+      this.styledHeader(messages.getMessage('actionRequired'));
+      this.log(messages.getMessage('enterCode', [loginData.user_code, loginData.verification_uri]));
+      this.log();
```

Comment:
these ux methods are no-ops in the json context, so they don't have to be carefully guarded.